File: repos_estudiantes_TARQUI_tercera_entrega/ILLANES_ESCOBAR_FERNANDO_JOHSUA/PERSISTENCIA/Ejercicio5/Python/gestor_farmacia.py
import pickle
import os
from sucursal import Sucursal

class GestorFarmacia:
    def __init__(self, archivo="farmacias.pkl"):
        self.archivo = archivo

    def guardar_sucursal(self, sucursal: Sucursal):
        lista = self.leer_todas()
        lista.append(sucursal)
        with open(self.archivo, 'wb') as f:
            pickle.dump(lista, f)

    def leer_todas(self):
        if not os.path.exists(self.archivo):
            return []
        with open(self.archivo, 'rb') as f:
            return pickle.load(f)

    def mostrar_medicamentos_tos(self, numero):
        for s in self.leer_todas():
            if s.numero == numero:
                print(f"Medicamentos para la tos - Sucursal {s.numero}")
                for m in s.medicamentos:
                    if m.tipo.lower() == "tos":
                        print(f"- {m.nombre}")

    def mostrar_sucursales_con_golpex(self):
        for s in self.leer_todas():
            for m in s.medicamentos:
                if m.nombre.lower() == "golpex":
                    print(f"Golpex se encuentra en: Sucursal {s.numero} - {s.direccion}")

File: repos_estudiantes_TARQUI_tercera_entrega/ILLANES_ESCOBAR_FERNANDO_JOHSUA/PERSISTENCIA/Ejercicio5/Java/GestorFarmacia.java
package PERSISTENCIA.Ejercicio5.Java;

import java.io.*;
import java.util.ArrayList;

public class GestorFarmacia {
    private final String archivo = "farmacias.dat";

    public void guardarSucursal(Sucursal s) {
        ArrayList<Sucursal> lista = leerTodas();
        lista.add(s);
        try (ObjectOutputStream out = new ObjectOutputStream(new FileOutputStream(archivo))) {
            out.writeObject(lista);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public ArrayList<Sucursal> leerTodas() {
        try (ObjectInputStream in = new ObjectInputStream(new FileInputStream(archivo))) {
            return (ArrayList<Sucursal>) in.readObject();
        } catch (IOException | ClassNotFoundException e) {
            return new ArrayList<>();
        }
    }

    public void mostrarMedicamentosTos(int numeroSucursal) {
        for (Sucursal s : leerTodas()) {
            if (s.getNumero() == numeroSucursal) {
                System.out.println("Medicamentos para la tos - Sucursal " + s.getNumero());
                for (Medicamento m : s.getMedicamentos()) {
                    if (m.getTipo().equalsIgnoreCase("tos")) {
                        System.out.println("- " + m.getNombre());
                    }
                }
            }
        }
    }

    public void mostrarSucursalesConGolpex() {
        for (Sucursal s : leerTodas()) {
            for (Medicamento m : s.getMedicamentos()) {
                if (m.getNombre().equalsIgnoreCase("Golpex")) {
                    System.out.println("Golpex se encuentra en: Sucursal " +
                        s.getNumero() + " - " + s.getDireccion());
                }
            }
        }
    }
}

